### PR TITLE
Fixes eager conversion on GPU

### PR DESCRIPTION
**Pull request by CYHSM**
Reopening from https://github.com/Modalities/modalities/pull/392 now into main branch.  See original below:

# What does this PR do?

This PR fixes the conversion error we get when running the conversion in eager mode on GPU. It seems the upcasting to float32 with subsequent downcasting introduces errors into the attention_weights and therefore also the logits (see below plot). So it seems that historically this was added when using half-precision (fp16) as it has limited range and can easily overflow/underflow, which should be fixed with the extended range of BF16.

comparison_differences

The logits still are similar enough to produce similar outputs so thats why we see similar outputs even though the logits of the converted model are not the same
comparison_logits

Not confirmed on main due to import errors from the updated transformers library, so comparing to the updating_and_fixing_conversion branch instead. 

## General Changes
* Changes the calculation of attn_weights to not upcast to float 32

## Checklist before submitting final PR
- [x] My PR is minimal and addresses one issue in isolation
- [x] I have merged the latest version of the target branch into this feature branch
- [x] I have reviewed my own code w.r.t. correct implementation, missing type hints, proper documentation, etc.
- [ ] I have run a sample config for model training
- [ ] I have checked that all tests run through (`python tests/tests.py`)
- [ ] I have updated the internal changelog (`CHANGELOG_DEV.md`)

### CODE REVIEW

**Review by le1nux — APPROVED** [review]
LGTM :)
